orca-pipelinetemplate/src/test/resources/integration/v1schema/loops-template.yml

```diff
@@ -0,0 +1,42 @@
+---
+schema: "1"
+id: loops
+metadata:
+  name: Loops
+  description: Stage and partial looping with variables
+variables:
+- name: waitTimes
+  description: A list of wait times
+  type: list
+stages:
+- id: loopedPartial
+  type: partial.waitGraph
+  config:
+    waitTime: "{{ templateLoop.value }}"
+  loopWith: "{{ waitTimes }}"
+- id: joinWait
+  type: wait
+  config:
+    waitTime: 3
+  dependsOn:
+  - loopedPartial
+- id: stageLoop
+  type: wait
+  name: "Wait {{ templateLoop.value }} seconds"
+  config:
+    waitTime: "{{ templateLoop.value }}"
+  loopWith: "{{ waitTimes }}"
```

Comment:
I think this is a slick idea, but would complicate the implementation quite a lot, potentially introducing the case where a stage would be looped over with multiple variables (thereby creating a matrix loop). Since my goal is to keep large changes to a minimum with MPT, I don't think I want to take on this level of change.

Reply:
Agreed, we would need to handle matrix loops in that case. I'm down with KISS'ng it.

Reply:
👄